src/event/flatten.rs

```diff
@@ -9,7 +10,7 @@ pub fn flatten(event: &mut LogEvent, map: Map<String, Value>) {
 }
 
 /// Recursevly inserts json values to event under given name
-pub fn insert(event: &mut LogEvent, name: String, value: Value) {
+pub fn insert<K: Into<Atom> + AsRef<str>>(event: &mut LogEvent, name: K, value: Value) {
```

Comment:
Can you explain why we need to `AsRef` here, `Into<..>` should work just fine? https://docs.rs/string_cache/0.8.0/string_cache/struct.Atom.html#impl-From%3C%26%27a%20str%3E

Reply:
In short, for optimization. 

It allows passing in `String` and `str` as name, which is useful for cases of `Value::Array` and `Value::Object` where Atom is not needed, and so having a way to bypass conversion to Atom is more performant. While at the same time, having an option to pass `Atom` as name because you already have it, avoids chain of conversions `Atom->String->Atom` in other `Value` cases as it was before.

Reply:
Not really a big fan of having two traits here but its fine for now.